storm-core/src/jvm/org/apache/storm/daemon/supervisor/SyncSupervisorEvent.java

```diff
@@ -366,14 +386,29 @@ protected boolean didDownloadBlobsForTopologySucceed(String stormconfPath, Strin
                 blobFileNames.add(ret);
             }
         }
-        for (String string : blobFileNames) {
-            if (!Utils.checkFileExists(string))
-                return false;
-        }
-        return true;
+
+        return didDownloadBlobsForTopologySucceed(targetDir, blobFileNames);
     }
 
     /**
+     * Assert if all dependencies blobs are downloaded for the given topology
+     * 
+     * @param stormcodePath
+     * @param targetDir
+     * @return
+     */
+    protected boolean didDownloadDependenciesForTopologySucceed(String stormcodePath, String targetDir) throws IOException {
```

Comment:
`areDependencyFilesDownloaded` would be fine for me since we're commenting like that.
Personally `check*` feels assertion which throws some exceptions if predicate doesn't hold true, but it's just me.